### PR TITLE
Compatible with pecl-mongodb

### PR DESCRIPTION
**Pull request by i0**
Compatible with pecl-mongodb for PHP 7 and earlier versions,
pecl-mongo package has deprecated and isn't available for PHP 7

### CODE REVIEW

File: src/Jenssegers/Mongodb/Relations/BelongsToMany.php [hunk 1]
```diff
@@ -126,9 +126,7 @@ public function sync($ids, $detaching = true)
         {
             $this->detach($detach);
 
-             $changes['detached'] = (array) array_map(function ($v) {
-                return is_numeric($v) ? (int) $v : (string) $v;
-            }, $detach);
+            $changes['detached'] = (array) array_map(function ($v) { return (int) $v; }, $detach);
```
Comment (jenssegers) [inline src/Jenssegers/Mongodb/Relations/BelongsToMany.php]: Why was this changed?

Comment (i0) [inline src/Jenssegers/Mongodb/Relations/BelongsToMany.php]: I didn't change it,
Probably my laravel-mongodb package was older.

File: src/Jenssegers/Mongodb/Model.php [hunk 2]
```diff
@@ -202,11 +201,11 @@ protected function getDateFormat()
...
      * Get a fresh timestamp for the model.
      *
-     * @return MongoDate
+     * @return MongoDB\BSON\UTCDateTime
      */
     public function freshTimestamp()
-    {
-        return new MongoDate;
+    {	
+	    return round(microtime(true) * 1000);
```
Comment (jenssegers) [inline src/Jenssegers/Mongodb/Model.php]: Shouldn't this return a UTCDateTime instance?

Comment (i0) [inline src/Jenssegers/Mongodb/Model.php]: In this case, it gives an exception because that the object values are not same between MongoDate and UTCDateTime, but the script will change generated microtime to an UTCDateTime instance later.

**Comment by jenssegers** [comment]
Do you know if there is a way of telling mongodb to return arrays instead of objects?

**Comment by i0** [comment]
There is a method named "toArray" for MongoDB\Driver\Cursor
see: http://php.net/manual/en/mongodb-driver-cursor.toarray.php

**Comment by Symfomany** [comment]
You must use MongoDB\BSON\UTCDateTime for handle datetime

**Comment by tobias-kuendig** [comment]
Make sure to rename the other `getName` and `ensureIndex` calls since they don't exist on https://github.com/mongodb/mongo-php-library anymore:

https://github.com/OFFLINE-GmbH/laravel-mongodb/compare/2dc68b03be0e9f9878937832029c16ca7d8fcf9d...OFFLINE-GmbH:ec2d90b26e6ec3fb3c2e24eea09ee873ef6666e8

**Comment by austinh** [comment]
Any update on this?

**Comment by andijames** [comment]
Yeah i'd be interested to know if there's any update too. If any help is required i'm more than happy to lend a hand where i can.

**Comment by qutwala** [comment]
+1, i'm too interested and waiting.

**Comment by jenssegers** [comment]
These changes are on the pecl-mongodb branch; but not all tests are green yet.
